### PR TITLE
Added support to Multiplexer and Manager to terminate durable workers

### PR DESCRIPTION
**Pull request by john-woolley**
I had a usecase for this in a project I'm working on and I thought it might be useful.  Happy to make any changes to be more in-line with the existing codebase or planned trajectory.

### CODE REVIEW

File: sanic/worker/manager.py [hunk 1]
```diff
@@ -408,6 +408,23 @@ def remove_worker(self, worker: Worker) -> None:
...
+    def terminate_durable_worker(self, ident: str) -> None:
+        """Terminate a durable worker and its processes.
+
+        Args:
+            ident (str): The name of the durable process to terminate.
+        """
+        worker = self.durable.get(ident)
+        if not worker:
+            error_logger.error(
+                f"Durable worker termination failed because {ident} not found."
```
Comment (ahopkins) [inline sanic/worker/manager.py:420]: Should log with `%s` and not f-string.

File: sanic/worker/multiplexer.py [hunk 2]
```diff
@@ -146,6 +146,14 @@ def terminate(self, early: bool = False):
         message = "__TERMINATE_EARLY__" if early else "__TERMINATE__"
         self._monitor_publisher.send(message)
 
+    def terminate_worker(self, ident: str):
+        """Terminate a worker.
+
+        Args:
+            name (str): The name of the worker to terminate.
+        """
```
Comment (prryplatypus) [inline sanic/worker/multiplexer.py:154]: `name` vs. `ident`

**Comment by john-woolley** [comment]
``=================================== FAILURES ===================================
________________________ test_terminate_durable_worker _________________________

manager = <sanic.worker.manager.WorkerManager object at 0x7f4f58e75190>
caplog = <_pytest.logging.LogCaptureFixture object at 0x7f4f58f30910>

    def test_terminate_durable_worker(manager: WorkerManager, caplog):
        worker = manager.manage("TEST", fake_serve, kwargs={})
    
        assert "Sanic-TEST-0" in worker.worker_state
        assert len(manager.transient) == 1
        assert len(manager.durable) == 1
    
        manager.terminate_durable_worker("WRONG")
        message = "Durable worker termination failed because WRONG not found."
    
        assert "Sanic-TEST-0" in worker.worker_state
        assert len(manager.transient) == 1
        assert len(manager.durable) == 1
>       assert ("sanic.error", 40, message) in caplog.record_tuples
E       AssertionError: assert ('sanic.error', 40, 'Durable worker termination failed because WRONG not found.') in [('sanic.error', 40, 'Durable worker termination failed (None not found).')]
E        +  where [('sanic.error', 40, 'Durable worker termination failed (None not found).')] = <_pytest.logging.LogCaptureFixture object at 0x7f4f58f30910>.record_tuples

/home/runner/work/sanic/sanic/tests/worker/test_manager.py:433: AssertionError

File: sanic/worker/manager.py [hunk 3]
```diff
@@ -408,6 +408,23 @@ def remove_worker(self, worker: Worker) -> None:
...
+    def terminate_durable_worker(self, ident: str) -> None:
+        """Terminate a durable worker and its processes.
+
+        Args:
+            ident (str): The name of the durable process to terminate.
+        """
+        worker = self.durable.get(ident)
+        if not worker:
+            error_logger.error(
+                "Durable worker termination failed (%s not found).", worker
```
Comment (prryplatypus) [inline sanic/worker/manager.py:420]: Should be `, ident`; not `, worker`. That seems to be what's making the test fail.

**Pull request closed without merging**